pkg/resource/appworker_deployment_builder.go

```diff
@@ -0,0 +1,132 @@
+// Licensed to Alexandre VILAIN under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Alexandre VILAIN licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package resource
+
+import (
+	"fmt"
+
+	"github.com/alexandrevilain/temporal-operator/api/v1beta1"
+	"github.com/alexandrevilain/temporal-operator/internal/metadata"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/utils/pointer"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+)
+
+type AppWorkerDeploymentBuilder struct {
+	instance *v1beta1.TemporalAppWorker
+	scheme   *runtime.Scheme
+}
+
+func NewAppWorkerDeploymentBuilder(instance *v1beta1.TemporalAppWorker, scheme *runtime.Scheme) *AppWorkerDeploymentBuilder {
+	return &AppWorkerDeploymentBuilder{
+		instance: instance,
+		scheme:   scheme,
+	}
+}
+
+func (b *AppWorkerDeploymentBuilder) Build() (client.Object, error) {
+	return &appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:        b.instance.ChildResourceName("appworker"),
```

Comment:
If I call my `TemporalAppWorker` `stock-service` it will become `stock-service-appworker`.
Maybe `worker` would be sufficient ?